tools/FhirSchemaManager/Commands/ApplyCommand.cs

```diff
@@ -4,16 +4,102 @@
 // -------------------------------------------------------------------------------------------------
 
 using System;
+using System.Collections.Generic;
+using System.CommandLine.Rendering;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using FhirSchemaManager.Exceptions;
+using FhirSchemaManager.Model;
+using FhirSchemaManager.Utils;
 
 namespace FhirSchemaManager.Commands
 {
     public static class ApplyCommand
     {
-        public static void Handler(string connectionString, Uri fhirServer, int version)
+        public static async Task HandlerAsync(string connectionString, Uri fhirServer, int version)
         {
-            Console.WriteLine($"--connection-string {connectionString}");
-            Console.WriteLine($"--fhir-server {fhirServer}");
-            Console.WriteLine($"--version {version}");
+            ISchemaClient schemaClient = new SchemaClient(fhirServer);
+
+            var region = new Region(
+                          0,
+                          0,
+                          Console.WindowWidth,
+                          Console.WindowHeight,
+                          true);
+
+            try
+            {
+                List<AvailableVersion> availableVersions = await schemaClient.GetAvailability();
+
+                if (availableVersions?.Any() != true || availableVersions.Count == 1)
+                {
+                    CommandUtils.PrintError(Resources.AvailableVersionsDefaultErrorMessage);
+                    return;
+                }
+                else
+                {
+                    // Removing the current version
+                    availableVersions.RemoveAt(0);
+                }
+
+                availableVersions = availableVersions.Where(availableVersion => availableVersion.Id <= version)
+                    .ToList();
+
+                foreach (AvailableVersion availableVersion in availableVersions)
+                {
+                    string script = await schemaClient.GetScript(availableVersion.Script);
+
+                    await ValidateVersion(schemaClient, availableVersion.Id);
+
+                    // check if the record for given version exists in failed status
+                    SchemaDataStore.ExecuteQuery(connectionString, string.Join(SchemaDataStore.DeleteQuery, availableVersion.Id, SchemaDataStore.Failed), availableVersion.Id);
```

Comment:
Should this be Format instead of Join?
`string.Join(SchemaDataStore.DeleteQuery, availableVersion.Id, SchemaDataStore.Failed)`.

A better way would be with sql parameters, example here: https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlcommand.parameters?view=netframework-4.8#examples